app/views/orders/index.html.erb

```diff
@@ -0,0 +1,70 @@
+<h1> Your Order Fulfillment Page</h1>
+
+<% total_revenue = 0 %>
+
+
+<div class="btn-group">
+  <button type="button" class="btn btn-primary dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+  Filter Orders by Status <span class="caret"></span>
+  </button>
+    <ul class="dropdown-menu">
+      <li><%= link_to "Paid Status", dashboard_orders_path(status: "paid") %></li>
+      <li><%= link_to "Complete Status", dashboard_orders_path(status: "complete") %></li>
+      <li><%= link_to "All Orders", dashboard_orders_path %></li>
+    </ul>
+  </div>
+
+
+
+<% @orders.each do |order|  %>
+<div class="row">
+  <div class="col-md-10">
+  <h3>Order #<%= link_to order.id, dashboard_order_show_path(:order_id => order.id) %></h3>
+  <h4>Status: <%= order.status.capitalize %></h4>
+  <h4>Order Placed: <%= order.created_at.localtime.strftime("%B %-d, %Y %I:%M %p") %></h4>
+
+  <table class="table">
+    <tr>
+      <th>Product Name</th>
+      <th>Price</th>
+      <th></th>
+      <th>Quantity</th>
+      <th>Subtotal</th>
+      <th>Shipped?</th>
+      <th>Mark as Shipped</th>
+    </tr>
+
+    <% order.order_items.each do |order_item| %>
+    <tr>
+      <% total_revenue += order_item.quantity * order_item.product.price  %>
+      <td><%=link_to order_item.product.name, product_path(order_item.product.id) %></td>
+      <td>$ <%= order_item.product.price %></td>
+      <td>x</td>
+      <td><%= order_item.quantity %></td>
+      <td><%= order_item.product.price * order_item.quantity %></td>
```

Comment:
We took a look at this section, and we all agreed this could use some major refactoring. However, for now we've removed the logic into a view helper -- while it didn't really cut down on the code, it removed the math calculations from the code.